src/treemodel/liverange.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+CKEDITOR.define( [
+	'treemodel/range',
+	'treemodel/liveposition',
+	'emittermixin',
+	'utils'
+], ( Range, LivePosition, EmitterMixin, utils ) => {
+	/**
+	 * LiveRange is a Range in the Tree Model that updates itself as the tree changes. It may be used as a bookmark.
+	 * **Note:** Be very careful when dealing with LiveRange. Each LiveRange instance bind events that might
+	 * have to be unbound. Use {@link #detach} whenever you don't need LiveRange anymore.
```

Comment:
Recently I changed `Range` constructor so it creates it's own `Position` instances from the passed variables. So you can create a range using `LivePosition`s but positions of newly created range won't be live.

This approach has much more sense when it comes to creating `LiveRange` and `Selection`. They create their own "live" parts, manage them and detach.